license_sh/helpers.py

```diff
@@ -315,6 +318,29 @@ def get_dependency_tree_with_licenses(
     return dependency_tree, unknown_licenses, has_issues
 
 
+def override_packages(
+    dep_tree: PackageNode,
+    overriden_packages: LanguageOverridenPackages,
+    analyze: bool = False,
+) -> PackageNode:
+    for node in PreOrderIter(dep_tree):
+        new_license = None
+        new_license_text = None
+        node_name, node_version = getattr(node, 'name', None), getattr(node, 'version', None)
```

Comment:
I suggest you write a simple pure function that will take anytree Node and `LanguageOverridenPackages` and do the transformation & write tests for it. 

After it's done, we can just call it within this loop.

+ I think that this `new_lincese` + `new_license_text` is a bit non-readable. What we need is a simple logic:

1) Make a set of overriden_packages
2) For that node, check whether it's in this set
3) If so, set corresponding fields + set a flag that we've used overriden license + reason.